nilearn/input_data/nifti_maps_masker.py

```diff
@@ -2,22 +2,62 @@
 Transformer for computing ROI signals.
 """
 
-import numpy as np
-
 from sklearn.base import BaseEstimator, TransformerMixin
 from sklearn.externals.joblib import Memory
 
 from .. import _utils
 from .._utils import logger
 from .._utils import CacheMixin
-from .._utils import new_img_like
-from .._utils.niimg_conversions import _check_same_fov
-from .._utils import _compose_err_msg
+from .._utils.cache_mixin import cache
+from .._utils.niimg_conversions import _check_same_fov, _assert_same_fov
 from .. import signal
 from .. import region
 from .. import image
 
 
+def _extract_signals(imgs, maps_img, smoothing_fwhm,
```

Comment:
Yes, but this is not the point of this PR. This is not as trivial as it seems. In particular, this is not the only code that can be factorized. It can be merged with the code coming from NiftiSpheresMasker and Searchlight (which have very similar code too). I was thinking of a BaseElementMasker that iterates on some kind of ROI iterator (list of ROI or labels or maps). But this require a lot of work. This is the step after!